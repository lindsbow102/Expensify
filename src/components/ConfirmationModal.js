import React from 'react';
import Modal from 'react-modal';

const ConfirmationModal = props => (
    <Modal
      isOpen={props.modalOpen}
      onRequestClose={props.closeModal}
      contentLabel="Remove Expense"
      closeTimeoutMS={200}
      className="remove--modal"
      ariaHideApp={false}
      modalAction={props.modalAction}
    >
      <h3 className="remove--modal__title">Are You Sure You Want To Remove This Expense?</h3>
      <button className="button button--danger" onClick={props.modalAction}>
        Remove
      </button>
      <button className="button button--secondary" onClick={props.closeModal}>
        Cancel
      </button>
    </Modal>
  );
  
  export default ConfirmationModal;