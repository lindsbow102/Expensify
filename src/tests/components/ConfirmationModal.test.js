import React from 'react';
import { shallow } from 'enzyme';
import ConfirmationModal from '../../components/ConfirmationModal';

test('should render Confirmation Modal correctly', () => {
    const wrapper = shallow(<ConfirmationModal />);
    expect(wrapper).toMatchSnapshot();
});