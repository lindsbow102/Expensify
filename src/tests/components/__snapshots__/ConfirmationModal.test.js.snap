// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Confirmation Modal correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="remove--modal"
  closeTimeoutMS={200}
  contentLabel="Remove Expense"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="remove--modal__title"
  >
    Are You Sure You Want To Remove This Expense?
  </h3>
  <button
    className="button button--danger"
  >
    Remove
  </button>
  <button
    className="button button--secondary"
  >
    Cancel
  </button>
</Modal>
`;
